### PR TITLE
Refactor documentation

### PR DESCRIPTION
**Pull request by fabratu**
This PR aims at refactoring Python/C++ documentation into one coherent build and fixes errors.

- Fixes sphinx 3.2.0 documentation build errors
- Fixes embedded rendering of notebooks
- Fixes small styling issues
- To discuss: Added (possibly) missing python modules. 
- To discuss: Uses [breathe ](https://breathe.readthedocs.io/en/latest/)and [exhale](https://exhale.readthedocs.io/en/latest/index.html) to merge documentation to sphinx-only build

Python doc (previously missing doc-modules):
`base`, `clique`, `coarsening`, `flow`, `GEXFIO` , `GraphMLIO`, `helpers`, `independentset`, `matching`, `partitioning`, `plot`, `randomization`, `sampling`, `scd`, `simulation`, `stats`, `traversal`, `workflows`
 
All python modules are now part of the documentation. `stats.pyx` und `profiling/stat.pyx` are merged to one module.

In order to recreate the proposed doc-structure locally, install `pip3 install sphinx sphinx_bootstrap_theme numpydoc breathe exhale nbsphinx sphinx_copybutton sphinxcontrib.bibtex sphinx_gallery sphinx_last_updated_by_git ipykernel ipython matplotlib nbconvert jupyter-client networkx tabulate` (majority of plugins used for notebook-rendering) and best follow the instructions from [travis-script](https://github.com/networkit/networkit/blob/master/.travis.yml#L324)

### CODE REVIEW

File: networkit/graphio.pyx [hunk 1]
```diff
@@ -507,7 +507,7 @@ cdef class BinaryPartitionWriter:
 	Partition ids are unsigned integers.
 
 	Parameters
```
Comment (cndolo) [inline networkit/graphio.pyx:509]: Seems like a colon is missing here.

File: networkit/centrality.pyx [hunk 2]
```diff
@@ -1917,15 +1917,15 @@ def relativeRankErrors(rx, ry):
...
+	-----------
 	rx : list
 		ranking - ordered list of tuples (node, score)
 
 	ry:  list
 		ranking - ordered list of tuples (node, score)
 
-	Returns
+	Returns:
 	_______
```
Comment (cndolo) [inline networkit/centrality.pyx:1929]: Previously (e.g L 1896) you replaced a line with dashes, should that be the case here as well?

File: networkit/distance.pyx [hunk 3]
```diff
@@ -938,8 +938,8 @@ cdef class DynAPSP(APSP):
 
 		Computes all pairwise shortest-path distances in G.
 
-		Parameters
-	----------
+		Parameters:
+	-----------
```
Comment (cndolo) [inline networkit/distance.pyx:942]: Maybe you could fix the indentation here while you're at it anyway.

**Comment by fabratu** [comment]
Thanks for the feedback. Took care of the issues.

Even though some files are not changed, the documentation build now runs fine. For example `randomization.pyx` has the "old" style, but seems not to be included in the documentation at all. Will add more commits in order to provide a more thorough fix.

**Comment by fabratu** [comment]
Added the missing documentation modules and changed the scope of the PR. Maybe some/all of them were intentionally not in the doc.

**Comment by fabratu** [comment]
I think the PR is ready now for review.

**Comment by avdgrinten** [comment]
LGTM, at least for an initial version. It seems that Exhale has quite a few limitations that we need to work around, maybe it would be a good idea to work with Exhale's upstream and post a PR to their repo to properly fix these issues.

**Review by avdgrinten — APPROVED** [review]

**Comment by fabratu** [comment]
Yeah, thats true - maybe exhale takes to much flexibility in favor of automation in the current state. Will look into that.

I am merging this earlier.

**Pull request merged**